### PR TITLE
Update druid API for simplifying YAML

### PR DESCRIPTION
**Pull request by tapojit047**

### CODE REVIEW

File: apis/kubedb/v1alpha2/druid_helpers.go [hunk 1]
```diff
@@ -419,6 +458,10 @@ func (d *Druid) SetDefaults() {
 				d.setDefaultContainerResourceLimits(&d.Spec.Topology.Historicals.PodTemplate, DruidNodeRoleHistoricals)
 			}
 		}
+
+		if d.Spec.Topology.Brokers == nil {
+			d.Spec.Topology.Brokers = &DruidNode{}
+		}
```
Comment (ArnobKumarSaha) [inline apis/kubedb/v1alpha2/druid_helpers.go:464]: No storage needed for nodes other than historicals & middlemanagers ?

File: apis/kubedb/v1alpha2/druid_types.go [hunk 2]
```diff
@@ -122,15 +118,15 @@ type DruidSpec struct {
 }
 
 type DruidClusterTopology struct {
-	Coordinators *DruidNode `json:"coordinators"`
+	Coordinators *DruidNode `json:"coordinators,omitempty"`
```
Comment (ArnobKumarSaha) [inline apis/kubedb/v1alpha2/druid_types.go:121]: Use `// +kubebuilder:default=1` in DruidNode.Replicas

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

File: apis/kubedb/v1alpha2/druid_helpers.go (hunk 1 above)
Comment (tapojit047) [reply apis/kubedb/v1alpha2/druid_helpers.go:452]: No, only for data node i.e. historicals and middleManagers

File: temp.json [hunk 3]
```diff
@@ -0,0 +1,59 @@
+{
```
Comment (ArnobKumarSaha) [inline temp.json:1]: Delete this

Comment (ArnobKumarSaha) [reply temp.json:1]: https://github.com/kubedb/apimachinery/blob/master/apis/kubedb/v1alpha2/pgbouncer_types.go#L215
Also remove this field & run `make gen`

**Review by ArnobKumarSaha — CHANGES_REQUESTED** [review]

**Review by ArnobKumarSaha — APPROVED** [review]

**Pull request merged**